configure.py

```diff
@@ -753,7 +753,7 @@ def get_gas_version(cc):
 # quite prepared to go that far yet.
 def check_compiler(o):
   if sys.platform == 'win32':
-    if not options.openssl_no_asm:
+    if not options.openssl_no_asm and options.dest_cpu in ('x86', 'x64'):
```

Comment:
```suggestion
    if not options.openssl_no_asm or options.dest_cpu = 'arm64':
```
_Explicit is better then implicit_

BTW https://github.com/nodejs/node/issues/23913 might be related.

Reply:
NASM only supports x64 and x86, so I figured the expressive way to code that was to only check for it in those two cases. I feel that x86 and x64 are the explicit exceptions with extra requirements here, not ARM64.